src/lms-harmonizer/edfi_lms_harmonizer/scripts/postgresql/0010_harmonize_assignment.sql

```diff
@@ -0,0 +1,133 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+create or replace procedure lms.harmonize_assignment(_sourcesystem varchar(255), _namespace varchar(255))
+as $$
+begin
+
+    create extension if not exists "pgcrypto";
+
+	create temp table all_assignments as
+	select
+		lmsassignment.sourcesystemidentifier as assignmentidentifier,
+		sourcesystemdescriptor.descriptorid as lmssourcesystemdescriptorid,
+		lmsassignment.title,
+		assignmentcatdescriptor.descriptorid as assignmentcategorydescriptorid,
+		lmsassignment.assignmentdescription,
+		lmsassignment.startdatetime,
+		lmsassignment.enddatetime,
+		lmsassignment.duedatetime,
+		lmsassignment.maxpoints,
+		edfisection.sectionidentifier,
+		edfisection.localcoursecode,
+		edfisection.sessionname,
+		edfisection.schoolyear,
+		edfisection.schoolid,
+		lmsassignment.lastmodifieddate as assignment_last_modified_date,
+		lmssection.deletedat as section_deleted,
+		lmsassignment.deletedat as assignment_deleted
+ 	from lms.assignment lmsassignment
+		inner join lms.lmssection lmssection
+			on lmsassignment.lmssectionidentifier = lmssection.lmssectionidentifier
+
+		inner join edfi.section edfisection
+			on lmssection.edfisectionid = edfisection.id
+
+		inner join edfi.descriptor sourcesystemdescriptor
+			on sourcesystemdescriptor.codevalue = lmsassignment.sourcesystem
+
+		inner join lmsx.lmssourcesystemdescriptor
+		    on sourcesystemdescriptor.descriptorid  = lmssourcesystemdescriptor.lmssourcesystemdescriptorid
+
+		inner join edfi.descriptor assignmentcatdescriptor
+			on assignmentcatdescriptor.codevalue = lmsassignment.assignmentcategory
+			and assignmentcatdescriptor.namespace = 'uri://ed-fi.org/edfilms/AssignmentCategoryDescriptor/' || _sourcesystem
```

Comment:
Ugh, I had forgotten about this bizarre (to me) way of handling string concatenation.